app/models/environment.rb

```diff
@@ -0,0 +1,25 @@
+class Environment < ApplicationRecord
+  belongs_to :heritage
+  delegate :district, to: :heritage
+
+  scope :plains, -> { where(value_from: nil) }
+  scope :secrets, -> { where.not(value_from: nil) }
+
+  validates :name, uniqueness: {scope: :heritage_id}, presence: true
+  validates :value, presence: true, if: -> { value_from.blank? }
+  validates :value_from, presence: true, if: -> { value.blank? }
+
+
+  # If value_from is nil return it as-is
+  # if value_from is a relative name prefix /barcelona/district-name/
+  #   ECS resolves relative value_from as a SSM parameter in the same account and region
+  def namespaced_value_from
+    return nil if value_from.nil?
+    return value_from if value_from.start_with?("arn:aws:")
+
+    s = value_from
+    s = "/" + s unless s.start_with?("/")
+    s = "/barcelona/#{district.name}" + s unless s.start_with?("/barcelona/#{district.name}")
+    s
```

Comment:
Users need to specify full ARN if they want to use secrets manager. ECS automatically use SSM parameter store if value_from is **not** full ARN so full ARN is the only way to use Secrets Manager. does that make sense?

Reply:
Or are you suggesting to remove relative path (paths not starting with `/barcelona/`)?

Reply:
Ah, I see.

There are two sources for secrets. So absolute path is only way to use non-standard path of SSM parameter. So it's required.

Reply:
> So absolute path is only way to use non-standard path of SSM parameter.

This is not correct. you can use full ARN for SSM parameters e.g. `arn:aws:ssm:region:aws_account_id:parameter/parameter_name`

I'm still thinking whether adding relative path is good or not...

- Full ARN
  - this is needed to use Secrets Manager
- Absolute path (Specifying `/barcelona/district/...` in `barcelona.yml`)
  - Specifying absolute path in value_from works same as ECS, so this needs to be supported
- relative path (omitting `/barcelona/district`)
  - only under `/barcelona/district/` is allowed by [task execution role](https://github.com/degica/barcelona/pull/501/files#diff-17d66829599d7e30e5bc04be0aa103ffR125). So specifying other path does not work out-of-the-box (users need to add an extra permission to the task execution role)
  - So it is natural to me that Barcelona treats relative path as sub-path of `/barcelona/district`
  - **Cons** if you want to use SSM absolute path that does not start with `/barcelona`, then you have to use full ARN

Reply:
@essa what do you think? (about adding relative path)

Reply:
From a user's perspective, troubles with permission is the worst error because they can be detected after deployment.

How about providing three options explicitly separated by the keyword?

- value: non-secret plain value
- value_path: secret value under control of Barcelona ( will be granted by Barcelona automatically)
- value_from: secret value under user's control ( should be granted by the user manually)

```yaml
image_name: quay.io/degica/repo
environment:
# Plain environment variable
- name: RAILS_ENV
  value: production
# this is resolved as SSM parameter store "/barcelona/district-name/production/secret_key_base"
- name: SECRET_KEY_BASE
  value_path: production/secret_key_base
# You can also use SecretsManager by specifying full ARN for your secrets
- name: DATABASE_URL
  value_from: arn:aws:secretsmanager:region:aws_account_id:secret:secrets-name-a1b2c3
# value_from is compatible with task-definitions, so you can specify absolute path for SSM parameter in the same region of the container
- name: XXX_API_KEY
  value_from: /api_keys/xxx
```

Reply:
Oh thanks, that sounds like a good idea. I'll think about it deeper over the weekend

Reply:
OK added `ssm_path` (is there  any better name?) which is resolved to relative path to `/barcelona/district`.

- If both `ssm_path` and `value_from` are specified, `ssm_path` takes precedence
- `ssm_path` is not persisted. Instead, it is converted to absolute `value_from` before save.
- `value_from` is now equivalent to TaskDefinition's value_from

Looks very clean, @essa thanks for your input